src/hypothesis/core.py

```diff
@@ -956,6 +956,17 @@ def run_test_with_generator(test):
 
         original_argspec = getfullargspec(test)
 
+        if getattr(test, 'is_hypothesis_test', False):
```

Comment:
One thing to consider is where this should warn. This currently warns at the point of *definition*. I think maybe it should warn at the point where the test is actually run? We normally try to defer errors to that point, and I *think* all deprecation warnings currently show up there.

In general having things happen at definition time seems to produce confusing output in test runners (at least in pytest).

Reply:
@rboulton - to warn when the test is run, you can just move this down a few lines into the inner function `wrapped_test`.